bank/domain/types.go

```diff
@@ -0,0 +1,76 @@
+package domain
+
+import "time"
+
+type SessionStore interface {
+	FindSessionAvailableAt(sessionID string, time time.Time) (*Session, bool)
+	StartSession(time time.Time, u User) (*Session, error)
+	DeleteSession(sessionID string)
+	UpdateSession(sessionID string, instant time.Time)
+}
+
+type UserStore interface {
+	InsertUser(u User) error
+	GetUser(u User) (*User, error)
+}
+
+type AccountStore interface {
+	GetAccounts(userID string) (*[]Account, error)
+	GetAccount(userID string, curr string) (*Account, error)
+	InsertAccount(a Account) (string, error)
+	UpdateAccount(a Account) error
+	DeleteAccount(userID string, currency string) error
+	InsertHistory(accID string, transType string, curr string, amount float64) error
+	GetHistory(accID string) (*[]History, error)
+}
+
+type Session struct {
+	SessionID string    `bson:"sessionid"`
+	UserID    string    `bson:"userid"`
+	Expires   time.Time `bson:"expires"`
+}
+
+type User struct {
+	UserID   string `bson:"userid"`
+	Username string `bson:"username"`
+	Password string `bson:"password"`
+	Name     string `bson:"name"`
+	Email    string `bson:"email"`
+	Age      int    `bson:"age"`
+	Hash     []byte `bson:"hash"`
+}
+
+type Account struct {
+	AccountID       string  `bson:"accountid"`
+	UserID          string  `bson:"userid"`
+	Currency        string  `bson:"currency"`
+	Amount          float64 `bson:"amount"`
+	TransactionType string  `bson:"transactiontype"`
+}
+
+type History struct {
+	AccountID       string    `bson:"accountid"`
+	TransactionType string    `bson:"transactiontype"`
+	Currency        string    `bson:"currency"`
+	Amount          float64   `bson:"amount"`
+	Date            time.Time `bson:"date"`
+}
+
+type Renew struct {
```

Comment:
I'm not sure that you need such thing, as logout could terminate session at any time. If you don't want to add logout in this PR, then you can perform manually logout by removing cookie from the browser.